lib/auth/auto_detect.js

```diff
@@ -12,19 +13,22 @@ var ChromeExtensionFlow = require('./chrome_extension_flow');
  *     appropriate type could be determined.
  */
 function autoDetect() {
-  if (typeof(chrome) !== 'undefined' && chrome.runtime && chrome.runtime.id) {
-    if (chrome.tabs && chrome.tabs.create) {
+  if (Environment.isChrome() && Environment.getChromeRuntime() && 
+        Environment.getChromeRuntime().id) {
```

Comment:
Thoughts on what I've done here with getChromeRuntime().id? What type of balance should I strive for?